plugin/scality/plugin.go

```diff
@@ -0,0 +1,335 @@
+	SecretKey         string
+	Bucket            string
+	PathPrefix        string
+	SOCKS5Proxy       string
+}
+
+func (p ScalityPlugin) Meta() plugin.PluginInfo {
+	return plugin.PluginInfo(p)
+}
+
+func (p ScalityPlugin) Validate(endpoint plugin.ShieldEndpoint) error {
+	var (
+		s    string
+		err  error
+		fail bool
+	)
+
+	s, err = endpoint.StringValue("scality_host")
+	if err != nil {
+		ansi.Printf("@R{\u2717 s3_host              %s}\n", err)
+		fail = true
+	} else {
+		ansi.Printf("@G{\u2713 s3_host}              @C{%s}\n", s)
+	}
+
+	s, err = endpoint.StringValue("access_key_id")
+	if err != nil {
+		ansi.Printf("@R{\u2717 access_key_id        %s}\n", err)
+		fail = true
+	} else {
+		ansi.Printf("@G{\u2713 access_key_id}        @C{%s}\n", s)
+	}
+
+	s, err = endpoint.StringValue("secret_access_key")
+	if err != nil {
+		ansi.Printf("@R{\u2717 secret_access_key    %s}\n", err)
+		fail = true
+	} else {
+		ansi.Printf("@G{\u2713 secret_access_key}    @C{%s}\n", s)
+	}
+
+	s, err = endpoint.StringValue("bucket")
+	if err != nil {
+		ansi.Printf("@R{\u2717 bucket               %s}\n", err)
+		fail = true
+	} else {
+		ansi.Printf("@G{\u2713 bucket}               @C{%s}\n", s)
+	}
+
+	s, err = endpoint.StringValueDefault("prefix", DefaultPrefix)
+	if err != nil {
+		ansi.Printf("@R{\u2717 prefix               %s}\n", err)
+		fail = true
+	} else if s == "" {
+		ansi.Printf("@G{\u2713 prefix}               (none)\n")
+	} else {
+		ansi.Printf("@G{\u2713 prefix}               @C{%s}\n", s)
+	}
+
+	s, err = endpoint.StringValueDefault("socks5_proxy", "")
+	if err != nil {
+		ansi.Printf("@R{\u2717 socks5_proxy         %s}\n", err)
+		fail = true
+	} else if s == "" {
+		ansi.Printf("@G{\u2713 socks5_proxy}         (no proxy will be used)\n")
+	} else {
+		ansi.Printf("@G{\u2713 socks5_proxy}         @C{%s}\n", s)
+	}
+
+	tf, err := endpoint.BooleanValueDefault("skip_ssl_validation", DefaultSkipSSLValidation)
+	if err != nil {
+		ansi.Printf("@R{\u2717 skip_ssl_validation  %s}\n", err)
+		fail = true
+	} else if tf {
+		ansi.Printf("@G{\u2713 skip_ssl_validation}  @C{yes}, SSL will @Y{NOT} be validated\n")
+	} else {
+		ansi.Printf("@G{\u2713 skip_ssl_validation}  @C{no}, SSL @Y{WILL} be validated\n")
+	}
+
+	if fail {
+		return fmt.Errorf("scality: invalid configuration")
+	}
+	return nil
+}
+
+func (p ScalityPlugin) Backup(endpoint plugin.ShieldEndpoint) error {
+	return plugin.UNIMPLEMENTED
+}
+
+func (p ScalityPlugin) Restore(endpoint plugin.ShieldEndpoint) error {
+	return plugin.UNIMPLEMENTED
+}
+
+func (p ScalityPlugin) Store(endpoint plugin.ShieldEndpoint) (string, error) {
+	scal, err := getScalityConnInfo(endpoint)
+	if err != nil {
+		return "", err
+	}
+	client, err := scal.Connect()
+	if err != nil {
+		return "", err
+	}
+
+	path := scal.genBackupPath()
+	plugin.DEBUG("Storing data in %s", path)
+
+	minio.SetMaxPartSize(1024 * 1024 * 75)
+	// FIXME: should we do something with the size of the write performed?
```

Comment:
Seems incorrectly marked as a FIXME. Nothing is broken. More of a... TODO? Either way, maybe print it to output? What do you think?